pkg/check/parameter_desc.go

```diff
@@ -68,35 +68,43 @@ type HumanReadableParamDesc struct {
 	NegationAllowed *bool                    `json:"negationAllowed,omitempty"`
 	SubParameters   []HumanReadableParamDesc `json:"subParameters,omitempty"`
 	ArrayElemType   ParameterType            `json:"arrayElemType,omitempty"`
+	NestingLevel    int                      `json:"-"`
 }
 
 // HumanReadableFields returns a human-friendly representation of this ParameterDesc.
 func (p *ParameterDesc) HumanReadableFields() HumanReadableParamDesc {
-	out := HumanReadableParamDesc{
-		Name:        p.Name,
-		Type:        p.Type,
-		Description: p.Description,
-		Required:    p.Required,
-		Examples:    p.Examples,
-	}
+	var transform func(*ParameterDesc, int) HumanReadableParamDesc
+
+	transform = func(p *ParameterDesc, level int) HumanReadableParamDesc {
```

Comment:
Yes, you're right. This can be a private function.
Initially I did not want to expose `level` argument that's why decided to hide `transform()` in a closure.
Moved it out now.